contracts/okp4-cognitarium/src/msg.rs

```diff
@@ -28,6 +28,42 @@ pub enum ExecuteMsg {
         /// are subject to the limitations defined by the `limits` specified at contract instantiation.
         data: Binary,
     },
+
+    /// # DeleteData
+    /// Delete the data (RDF triples) from the store matching the patterns defined by the provided
+    /// query. For non-existing triples it acts as no-op.
+    ///
+    /// Example:
+    /// ```json
+    /// {
+    ///   "prefixes": [
+    ///     { "prefix": "foaf", "namespace": "http://xmlns.com/foaf/0.1/" }
+    ///   ],
+    ///   "delete": [
+    ///     { "variable": "person" },
+    ///     { "variable": "property" },
+    ///     { "variable": "value" }
+    ///   ],
+    ///   "where": [
+    ///     { "simple": { "triplePattern": {
+    ///         "subject": { "variable": "person" },
+    ///         "predicate": { "node": { "namedNode": {"prefixed": "foaf:givenName"} } },
+    ///         "object": { "literal": { "simple": "Myrddin" } }
+    ///    } } }
+    ///  ]
+    /// ```
+    ///
+    /// Only the smart contract owner (i.e. the address who instantiated it) is authorized to perform
+    /// this action.
+    DeleteData {
+        /// The prefixes used in the operation.
+        prefixes: Vec<Prefix>,
+        /// The items to delete.
+        delete: Vec<SelectItem>,
```

Comment:
Just to check I understand it well, we delete the triples linked to the `SelectItems`? For example If a variable is present in the `WHERE` clause, but not selected, the underlying variable's triples won't be deleted, is that correct?

Reply:
My bad, it's not correct. The item to delete should be `Vec<TriplePattern>`. This allows to specify the triples to delete by using a pattern with can refer to plain values or variables from the `where` clause.

Reply:
Ah this makes more sense, excellent! 😉